### PR TITLE
scons: SConscript/SConstruct refactor

### PR DESCRIPTION
**Pull request by fredyshox**
Test builds:
- [x] linux x86_64
- [x] darwin
- [x] agnos aarch64
- [x] linux aarch64  

Other
- [x] test multithreaded build

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -64,53 +64,57 @@ AddOption('--no-test',
...
-## - aarch64: linux pc aarch64
-## - x86_64:  linux pc x64
-## - Darwin:  mac x64 or arm64
-real_arch = arch = subprocess.check_output(["uname", "-m"], encoding='utf8').rstrip()
-if platform.system() == "Darwin":
-  arch = "Darwin"
+# *** Target and Architecture
+
+## Target name breakdown (target)
+## - agnos-aarch64: linux tici aarch64
```
Comment (fredyshox) [inline SConstruct:70]: @adeebshihadeh what do you think about replacing slightly ambiguous `larch64` with `agnos-aarch64` for TICI target?

**Comment by fredyshox** [comment]
Submodules PRs
rednose: https://github.com/commaai/rednose/pull/34
opendbc: https://github.com/commaai/opendbc/pull/926
cereal: https://github.com/commaai/cereal/pull/524

**Comment by fredyshox** [comment]
@adeebshihadeh I've revived this one, could u take a look?

**Comment by haraschax** [comment]
There are too many things going on in this PR. There are renames, refactors, comment styling changes. Those should all be separated, especially when touching critical things like the constructs.

**Comment by fredyshox** [comment]
I agree. I see couple of things that could be separated:
- arch => target rename
- remove unused stuff (from c2 days)
- restructure and cleanup

**Pull request closed without merging**